### PR TITLE
Add getters to Message and make attributes private

### LINKED ISSUE #28: Message source and timestamp should be immutable after creation
`Message` struct's `source` and `timestamp` should be immutable after creation and it should not be possible to directly modify metrics or tags. We should remove public struct fields and expose only getter methods.

### PR DESCRIPTION
**Pull request by adarshk7**
Closes #28

### CODE REVIEW

File: src/message.rs [hunk 1]
```diff
@@ -23,6 +23,18 @@ impl Message {
     pub fn insert_data(&mut self, field: &str, value: impl Into<PayloadValue>) {
         self.payload.insert(String::from(field), value.into());
     }
+
+    pub fn get_payload(&self) -> &HashMap<String, PayloadValue> {
```
Comment (ollipa) [inline src/message.rs:27]: Getters should not have `get_` in the name. See https://doc.rust-lang.org/1.0.0/style/style/naming/README.html

File: src/message.rs [hunk 2]
```diff
@@ -23,6 +23,18 @@ impl Message {
...
+        &self.payload
+    }
+
+    pub fn get_source(&self) -> &String {
+        &self.source
+    }
+
+    pub fn get_timestamp(&self) -> DateTime<Utc> {
```
Comment (ollipa) [inline src/message.rs:35]: Why a reference is not returned here?

Comment (adarshk7) [reply src/message.rs]: Just something I forgot to do :D

File: src/message.rs (hunk 1 above)
Comment (adarshk7) [reply src/message.rs]: I was thinking the same, but didn't know if it would work but really cool that there's actually official guidelines for this!

**Comment by adarshk7** [comment]
@ollipa Fixed!

**Pull request merged**